Move AOI Upload Into Sidebar and Accept Zipfiles

arottersman opened the pull request:
## Overview

- Move shapefile upload to sidebar
- Wire up "Select file" button
- Instead of accepting `.shp`, take `.zip`
- To unzip the compressed shapefile, I added [`JSZip`](https://github.com/Stuk/jszip)
    - Unzipping files is non-trivial, so a library is probably best
    - The other option in the js package space is [`zip.js`](https://github.com/gildas-lormeau/zip.js) but it isn't available via npm, nor is it versioned

Connects #1760 

### Demo

![uploadfilemmw](https://cloud.githubusercontent.com/assets/7633670/25020908/345104aa-205e-11e7-9081-bb2017334c88.gif)

### Notes

The card mentions a loading spinner, but it's so instantaneous I'm not sure it's necessary to add it with this PR. 


## Testing Instructions

 * If it works`./scripts/npm.sh install`, or `ssh worker` then `cd /vagrant/src/mmw && npm install`
 * `./scripts/bundle.sh --vendor && ./scripts/bundle.sh`
 * Drag and drop some geojson and zipped shp files, and also add them via the "Select file" button
 * Zipfiles without a `.shp` should display an error modal

caseycesari commented on src/mmw/js/src/draw/templates/aoiUpload.html:
```diff
@@ -1 +1,21 @@
+                <li>Only the first feature is used.</li>
+            </ul>
+        <button class="btn btn-success pull-left"
+                onclick="document.getElementById('draw-tool-file-upload-input').click();">
```
Curious about this inline event binding. Is it not possible to do this in a Backbone view?

arottersman replied:
It is — I'll push a fixup commit

arottersman replied:
Having it in the template was somewhat nice because it is explicit about the button being a dummy for the hidden input. Writing the `onclick` code as a string is not ideal though

caseycesari replied:
Yeah, I think adding a note in the commit message about that might be helpful.

caseycesari commented:
Re: the loading spinner, I agree. In my tests, the files have been processed almost simultaneously since we don't have to do any actual uploading. Shapefiles are limited to 2gb by design, and since we are only using the first record, I think we should be able to handle any shapefile fine. GeoJSON has no file size limit, but if you upload GeoJSON with a complex shape, the browser hangs as the geometry is parsed. We may want to limit the size of GeoJSON (as a correlate, since we don't know if the geometry is complex until we process it). The browser receives the file metadata before it reads in the file. In any case, I think this could be handled in https://github.com/WikiWatershed/model-my-watershed/issues/1761 or another card.

caseycesari commented:
Two more things that we talked about, that are for other cards, but that I wanted to make a note of so we didn't forget:

- We may want to add a mask when processing the file, just like geojson.io, to give an indication that the UI is locked while the file is being processed.
- GeoJSON.io seems to parse geojson files way faster than us. We should take a look at what they are doing.

caseycesari commented on src/mmw/js/src/draw/templates/aoiUpload.html:
```diff
@@ -1 +1,22 @@
+            Drag and drop a file
+            <ul class="draw-pane-description-list">
+                <li>Must be a shapefile (.zip) or geojson</li>
+                <li>Less than 200 MB</li>
```
I suggested this, but I think we should remove it until we actually implement a limit.

arottersman commented:
Thanks for taking a look! 

- Removed the message about the 200mb limit, but not the check (it's usefulness can be evaluated in #1761 )
- Squashed the fixup
